ts/components/session/ActionsPanel.tsx

```diff
@@ -88,6 +114,31 @@ const Section = (props: { type: SectionType; avatarPath?: string }) => {
     );
   }
 
+  let iconColor = undefined;
+  if (type === SectionType.PathIndicator) {
+    // Set icon color based on result
+    const red = theme.colors.destructive;
+    const green = theme.colors.accent;
+    const orange = theme.colors.warning;
+
+    iconColor = hasOnionPath ? theme.colors.accent : theme.colors.destructive;
+    const onionState = useSelector((state: StateType) => state.onionPaths);
+
+    iconColor = red;
+    const isOnline = useNetwork();
+    if (!(onionState && onionState.snodePath) || !isOnline) {
+      iconColor = Constants.UI.COLORS.DANGER;
+    } else {
+      const onionSnodePath = onionState.snodePath;
+      if (onionState && onionSnodePath && onionSnodePath.path.length > 0) {
+        let onionNodeCount = onionSnodePath.path.length;
+        iconColor = onionNodeCount > 2 ? green : onionNodeCount > 1 ? orange : red;
+      }
+    }
+  }
+
+  const unreadToShow = type === SectionType.Message ? unreadMessageCount : undefined;
+
```

Comment:
we should probably move that to another component which is rendered when  `SectionType.PathIndicator` is set